core/postInitClient.sqf

```diff
@@ -29,34 +29,111 @@
+            killcam_killer_pos = ASLtoAGL eyePos _killer;
+        } else {
+            //we will try to retrieve info from our hit EH
+            private _last_hit_info = [];
+            if (!isNil "killcam_LastHit") then {
+                _last_hit_info = killcam_LastHit;
+            };
+            
+            //hit info retrieved, now we check if it's not caused by fall damage etc.
+            //also we won't use info that's over 10 seconds old
+            if (count _last_hit_info != 0) then {
+                _last_hit_info params ["_data", "_time", "_unitPos", "_killerPos"];
+                if (_time + 10 > CBA_missionTime &&
+                {(_data select 1) != objNull} &&
+                {(_data select 1) != player}
+                ) then {
+                    killcam_unit_pos = _unitPos;
+                    killcam_killer = _data select 1;
+                    killcam_killer_pos = _killerPos;
+                }
+                else {
+                    //everything failed, we set value we will detect later
+                    killcam_killer_pos = [0,0,0];
+                    killcam_unit_pos = ASLtoAGL eyePos _unit;
+                    killcam_killer = objNull;
+                };
+            }
+            else {
+                killcam_killer_pos = [0,0,0];
+                killcam_unit_pos = ASLtoAGL eyePos _unit;
+                killcam_killer = objNull;
+            };
+        };
+    
+        if (eg_instant_death) then {
+            private _damage = 0.5;
+            if (!isNil "killcam_LastHitDamage") then {
+                _damage = killcam_LastHitDamage;
+            };
+            cutText ["\n", "BLACK", 0.1, true];
+            [QGVAR(death), 0, true] call ace_common_fnc_setHearingCapability;
+            0 fadeSound 0;
+            [{
+                ["<t color='#FF0000'>YOU ARE DEAD</t>", 0, 0.4, 2, 0.5, 0, 1000] spawn BIS_fnc_dynamicText;
+            }, [], 1] call CBA_fnc_waitAndExecute;
+        } else {
+            private _damage = 0.5;
+            if (!isNil "killcam_LastHitDamage") then {
+                _damage = killcam_LastHitDamage;
+            };
+            [{
+                params ["_damage"];
+                cutText ["\n", "BLACK", (1.01 - _damage), true];
+                [QGVAR(death), 0, true] call ace_common_fnc_setHearingCapability;
+                [{
+                    params ["_damage"];
+                    0 fadeSound 0;
+                    [{
+                        ["<t color='#FF0000'>YOU ARE DEAD</t>", 0, 0.4, 2, 0.5, 0, 1000] spawn BIS_fnc_dynamicText;
+                    }, [], (1.01 - _damage)] call CBA_fnc_waitAndExecute;
+                }, [_damage], (1.01 - _damage)] call CBA_fnc_waitAndExecute;
+            }, [_damage]] call CBA_fnc_execNextFrame;
+        };
+    }];
 	
 	if (GVAR(StartOnSafe)) then {
 		[{(!isNull ace_player)}, {
-			if (currentWeapon player != "") then {
+			if !(currentWeapon player isEqualTo "") then {
```

Comment:
Personally to make this more readable and confirm the order of precedence I'd do:
`!((currentWeapon player) isEqualTo "")` but whatever.

Reply:
order of precedence in sqf is always (), then unary over binary, then left to right. The standard for a while has been as few () as possible.

https://ace3mod.com/wiki/development/coding-guidelines.html#55-brackets-around-code

Reply:
I would argue that yes according to that standard in the effort of using fewer it has the side effect of decreasing readability but that's me. I don't really mind but personally I'd use brackets to improve the readability. I know the precedence is known but I find it's clearer and makes it easier to glance at an expression and quickly understand what it would do rather than having to mentally apply the precedence rules. It's your choice though.

Reply:
I would agree if it was a longer series of unary commands, but not for `currentweapon player` which is pretty obvious as to what the `isEqualTo` checks, it checks for an empty string for the other side of the equation, which is currentweapon player

Reply:
Yeah fair enough, as long as it's clear to the author is the main thing and I think it's clear to most others too.